fujiji-client/src/components/ListingForm/ListingForm.jsx

```diff
@@ -206,7 +206,24 @@ export default function ListingForm({
       setUploadErrorMessage('Allowed files: .png, .jpg, .jpeg');
     } else {
       setUploadedImage(uploadedFile);
-      setListingImageUrl(URL.createObjectURL(uploadedFile));
+      const url = onImageUpload(uploadedFile);
+      if (url.error) {
```

Comment:
`url.error` wouldn't exist here. Since url at this point is still a promise

because we don't `await` onImageUpload(uploadedFile).

Reply:
```
const url = await onImageUpload(uploadedFile);

if (url.error) {
   toast({...})
}
else {
    const imageUrl = url.data.imageUrl;
    set image()....
}
```